eddy/ui/ontology.py

```diff
@@ -1342,3 +1366,400 @@ def doApplyIriLabel(self):
+
+        diagramLayout = QtWidgets.QGridLayout(self)
+        diagramLayout.setContentsMargins(8, 8, 8, 8)
+
+        diagramLayout.addWidget(self.widget('add_annotations'))
+        diagramLayout.addWidget(self.widget('override_annotations'))
+
+
+        diagramGroup = QtWidgets.QGroupBox('Overriding Options', self)
+        diagramGroup.setLayout(diagramLayout)
+
+        diagramGroupLayout = QtWidgets.QHBoxLayout(self)
+        diagramGroupLayout.addWidget(diagramGroup)
+
+        diagramWidget = QtWidgets.QWidget(self)
+        diagramWidget.setLayout(diagramGroupLayout)
+
+        confirmation = QtWidgets.QDialogButtonBox(QtCore.Qt.Horizontal, self)
+        confirmation.addButton(QtWidgets.QDialogButtonBox.Ok)
+        confirmation.addButton(QtWidgets.QDialogButtonBox.Cancel)
+        confirmation.setObjectName('confirmation')
+        self.addWidget(confirmation)
+
+
+        buttonLayout = QtWidgets.QHBoxLayout(self)
+        buttonLayout.setAlignment(QtCore.Qt.AlignRight)
+        buttonLayout.addWidget(self.widget('confirmation'), 0, QtCore.Qt.AlignRight)
+
+        buttonWidget = QtWidgets.QWidget(self)
+        buttonWidget.setLayout(buttonLayout)
+
+        mainLayout = QtWidgets.QVBoxLayout()
+        mainLayout.setContentsMargins(10, 10, 10, 10)
+        mainLayout.addWidget(diagramWidget)
+        mainLayout.addWidget(buttonWidget)
+
+        self.setLayout(mainLayout)
+        self.setWindowIcon(QtGui.QIcon(':/icons/128/ic_eddy'))
+        self.setWindowTitle('Overriding Options')
+
+        connect(confirmation.accepted, self.accept)
+        connect(confirmation.rejected, self.reject)
+
+    #############################################
+    #   PROPERTIES
+    #################################
+
+    @property
+    def session(self):
+        """
+        Returns the active session (alias for self.parent()).
+        :rtype: Session
+        """
+        return self.parent()
+
+    @property
+    def project(self):
+        """
+        Returns the active project.
+        :rtype: Project
+        """
+        return self._project or self.session.project
+
+    #############################################
+    #   SLOTS
+    #################################
+
+    @QtCore.pyqtSlot()
+    def onOptionChecked(self):
+        """
+        Executed when an diagram checkbox is clicked.
+        """
+        option = self.sender().text()
+        state = self.sender().isChecked()
+
+        if state:
+            if option == 'Add annotations to the existing ones':
+
+                self.widget('override_annotations').setChecked(False)
+
+            else:
+
+                self.widget('add_annotations').setChecked(False)
+
+            self.widget('confirmation').setEnabled(True)
+
+        else:
+
+            self.widget('confirmation').setEnabled(False)
+
+
+
+    def checkedOption(self):
+
+        if self.widget('override_annotations').isChecked():
+            return True
+        else:
+            return False
```

Comment:
Questa logica sarebbe meglio fattorizzarla in uno (o più) OntologyLoaders, in modo da tenerla separata dal codice dell'interfaccia grafica